src/rest_vol_attr.c

```diff
@@ -2019,7 +2035,25 @@ RV_attr_specific(void *obj, const H5VL_loc_params_t *loc_params, H5VL_attr_speci
         RV_free(host_header);
 
     if (attr_iter_object_id >= 0) {
-        if (H5I_GROUP == parent_obj_type) {
+        if (H5I_FILE == parent_obj_type) {
+            /* Free copied fapl, fcpl, filepath, and object struct. */
+            /*
+            RV_object_t *attr_iter_obj = (RV_object_t*) attr_iter_object;
+
+            if (H5Pclose(attr_iter_obj->u.file.fapl_id) < 0)
+                FUNC_DONE_ERROR(H5E_PLIST, H5E_CANTCLOSEOBJ, FAIL, "can't close FAPL");
+            if (H5Pclose(attr_iter_obj->u.file.fcpl_id) < 0)
+                FUNC_DONE_ERROR(H5E_PLIST, H5E_CANTCLOSEOBJ, FAIL, "can't close FCPL");
+
+            if (attr_iter_obj->u.file.filepath_name)
+                RV_free(attr_iter_obj->u.file.filepath_name);
```

Comment:
Was this cleanup code intended to be commented out? Is it handled elsewhere?

Reply:
Yes, the cleanup in this case is handled by H5Fclose